### DISCUSSION TITLE
Explore RBest multivariate MAP priors

### DISCUSSION THREAD

**Discussion by wlandau** [openpharma/brms.mmrm#86] [Ideas]
https://github.com/Novartis/RBesT/issues/12

**Comment by weberse2** [comment]
Now on CRAN.

**Comment by wlandau** [comment]
Like #85, this is likely to be an application of #40 rather than a feature, so I am converting it into a discussion.

**Comment by wlandau** [comment]
c.f. https://opensource.nibr.com/bamdd/src/02l_single_arm_pos.html

**Reply by wlandau** [reply]
In the case of univariate mixtures, https://opensource.nibr.com/bamdd/src/02l_single_arm_pos.html#specification-of-prior-distributions helpfully describes the `prior(mixnorm())` syntax:

```r
brms_prior <- prior(class=b, coef=Intercept, normal(-1.67, 1.0)) + # prior average placebo effect 
  prior(class=sd, coef=Intercept, group=trial, lognormal( -1.06, 0.35)) + # prior on the between-study SD in  placebo log-odds
  prior(class=b, coef=long_study, normal(-0.1, 0.1)) + # prior for effect of long-study (26 vs. 12 weeks) on responder rate
  prior(class=b, coef="treatmentNVS101:long_study", normal(0, 0.1/1.96)) + # Prior on the log-odds-ratio for drug vs. placebo when switching from shor to long study
  prior(mixnorm(map_w, map_m, map_s), class=b, coef=treatmentNVS101) #  Syntax for specifying mixture prior
```

> NOTE: Thanks to the new functionality in the RBesT package, we were able to just write prior(mixture_prior, class=b, coef=treatmentNVS10) to specify a mixture prior. Alternatively, we could have written prior(mixnorm(map_w, map_m, map_s), class=b, coef=treatmentNovDrug). map_w, map_m and map_s would specify the weights, means and standard deviations of the mixture components, respectively. If we did not have this RBesT functionality available, we would have to manually write down the log-probability-density-function of the mixture, which involves weighted sums on the probability scale e.g. using the log_sum_exp function. For further details on the mixstanvars feature, please refer to its [online manual](https://opensource.nibr.com/RBesT/reference/mixstanvar.html).

**Reply by wlandau** [reply]
https://opensource.nibr.com/RBesT/reference/mixstanvar.html also describes an example that uses `stanvars` to inject Stan code.

**Comment by wlandau** [comment]
Thanks @weberse2 for implementing this in `RBesT`! Univariate MAP priors are a breeze. I wrote about the integration with `brms.mmrm` in https://github.com/openpharma/brms.mmrm/pull/138.

In that draft vignette, I also have a section at the bottom about multivariate mixtures. If you know of an easier way to set up multivariate MAP priors (`gMAP()` + `automixfit()` + `robustify()` instead of a custom `mixmvnorm()`) I would be interested.
